src/svd/two-sided/nsvd.cpp

```diff
@@ -0,0 +1,77 @@
+#include "nsvd.hpp"
+#include <math.h>
+#include "util.hpp"
+
+struct svd_2x2_params nsvd(double w, double x, double y, double z) {
+    bool flag = false;
+    double c = 0.0, c1 = 0.0, c2 = 0.0;
+    double s = 0.0, s1 = 0.0, s2 = 0.0;
+    double k = 1.0;
+
+    if (y == 0.0 && z == 0.0) {
```

Comment:
Do we pass 0.0 exactly as `y` and `z` parameters to this function? Otherwise, I don't see how this will evaluate to true.

Reply:
It is possible as the array we apply the svd on could contain the value 0. In the paper it is mentioned that the case where y and z are 0 need special handling. This code is based on the code that is provided in the paper.